pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -590,110 +738,158 @@ func shouldSkipUnmanagedStore(ctx context.Context, namespace string, r *Reconcil
 
-func shouldReconcile(es esv1beta1.ExternalSecret) bool {
-	if es.Spec.Target.Immutable && hasSyncedCondition(es) {
+// isSecretValid checks if the secret exists, and it's data is consistent with the calculated hash.
+func isSecretValid(existingSecret *v1.Secret) bool {
+	// if target secret doesn't exist, we need to refresh
+	if existingSecret.UID == "" {
 		return false
 	}
-	return true
-}
-
-func hasSyncedCondition(es esv1beta1.ExternalSecret) bool {
-	for _, condition := range es.Status.Conditions {
-		if condition.Reason == "SecretSynced" {
-			return true
-		}
-	}
-	return false
-}
 
-// isSecretValid checks if the secret exists, and it's data is consistent with the calculated hash.
-func isSecretValid(existingSecret v1.Secret) bool {
-	// if target secret doesn't exist, or annotations as not set, we need to refresh
-	if existingSecret.UID == "" || existingSecret.Annotations == nil {
+	// if the managed label is missing or incorrect, then it's invalid
+	if existingSecret.Labels[esv1beta1.LabelManaged] != esv1beta1.LabelManagedValue {
 		return false
 	}
 
-	// if the calculated hash is different from the calculation, then it's invalid
+	// if the data-hash annotation is missing or incorrect, then it's invalid
+	// this is how we know if the data has chanced since we last updated the secret
 	if existingSecret.Annotations[esv1beta1.AnnotationDataHash] != utils.ObjectHash(existingSecret.Data) {
 		return false
 	}
-	return true
-}
 
-// computeDataHashAnnotation generate a hash of the secret data combining the old key with the new keys to add or override.
-func (r *Reconciler) computeDataHashAnnotation(existing, secret *v1.Secret) string {
-	data := make(map[string][]byte)
-	maps.Insert(data, maps.All(existing.Data))
-	maps.Insert(data, maps.All(secret.Data))
-
-	return utils.ObjectHash(data)
+	return true
 }
 
 // SetupWithManager returns a new controller builder that will be started by the provided Manager.
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options) error {
 	r.recorder = mgr.GetEventRecorderFor("external-secrets")
 
-	// Index .Spec.Target.Name to reconcile ExternalSecrets effectively when secrets have changed
-	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &esv1beta1.ExternalSecret{}, externalSecretSecretNameKey, func(obj client.Object) []string {
+	// index ExternalSecrets based on the target secret name,
+	// this lets us quickly find all ExternalSecrets which target a specific Secret
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &esv1beta1.ExternalSecret{}, indexESTargetSecretNameField, func(obj client.Object) []string {
 		es := obj.(*esv1beta1.ExternalSecret)
-
-		if name := es.Spec.Target.Name; name != "" {
-			return []string{name}
+		// if the target name is set, use that as the index
+		if es.Spec.Target.Name != "" {
+			return []string{es.Spec.Target.Name}
+		} else {
+			// otherwise, use the ExternalSecret name
+			return []string{es.Name}
 		}
```

Comment:
```suggestion
		// if the target name is set, use that as the index
		if es.Spec.Target.Name != "" {
			return []string{es.Spec.Target.Name}
		}
		
		// otherwise, use the ExternalSecret name
		return []string{es.Name}
```

No need for the else inside the function closure.

Reply:
Usually if you return an else is superfluous. And yes, this is a practice in the codebase so I would rather keep the same practice everywhere.

Reply:
Also, elses add to the complexity score. Which we don't want to increase.

Reply:
Thank you